### PR TITLE
[TensorExpr] Re-enable tests.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44218 [TensorExpr] Re-enable tests.**
* #44270 Benchmarks: re-enable profiling-te configuration (try 2).

Differential Revision: [D23546100](https://our.internmc.facebook.com/intern/diff/D23546100)

### CODE REVIEW

File: test/test_jit_fuser_te.py [hunk 1]
```diff
@@ -860,25 +858,13 @@ def test_lstm_traced_cuda(self):
...
-            FileCheck.check("FusionGroup").run(str(graph))
-        except RuntimeError as e:
-            if 'Failed to compile' in e.args[0]:
-                warnings.warn('CPU fuser test has failed! This is not a hard failure, '
-                              'because the kernels sometimes trigger bugs in compilers '
-                              '(most notably GCC 7.2).')
-                raise unittest.SkipTest('Failed to compile')
-            else:
-                raise
+        ge = self.checkTrace(LSTMCellF, inputs)
```
Comment (ZolotukhinM) [inline test/test_jit_fuser_te.py:863]: I think the comment is only relevant to the old fuser.

File: test/test_tensorexpr.py [hunk 2]
```diff
@@ -1025,23 +1024,22 @@ def test_int(x, y, z, a, b):
...
-#            b = b + y
-#        return b
-#
-#    llvm = LLVMCodeGenExecuted()
-#    interp = SimpleIREvalExecuted()
-#    x, y, z = (torch.zeros(32, 32), torch.ones(32, 32), 4)
-#    test(x, y, z)
-#    r = test(x, y, z)
-#    assert llvm.elapsed_value == 1 or interp.elapsed_value() > 1
+    def test_loop(self):
```
Comment (ZolotukhinM) [inline test/test_tensorexpr.py:1027]: This seems to work now, so why not enable it?

File: test/test_tensorexpr.py [hunk 3]
```diff
@@ -1210,19 +1208,6 @@ def run_where(x, y):
         y = run_where(a, b)
         np.testing.assert_allclose(x.numpy(), y.numpy())
 
-    @unittest.skipIf(not torch.cuda.is_available(), "requires CUDA")
-    @unittest.skip("temporarily disable")
-    def test_unused(self):
```
Comment (ZolotukhinM) [inline test/test_tensorexpr.py:1215]: I don't understand what this test was checking and it seems that we have other tests for `rand_like`, so let's just remove this one (it works, but I think it's redundant).

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Pull request closed without merging**